mlflow/cli/traces.py

```diff
@@ -0,0 +1,738 @@
+            "info.execution_duration",
+            "info.request_preview",
+            "info.response_preview",
+        ]
+
+    if output == "json":
+        if field_list is None:
+            # Full JSON output
+            result = {
+                "traces": [trace.to_dict() for trace in traces],
+                "next_page_token": traces.token,
+            }
+        else:
+            # Custom fields JSON output - filter original structure
+            traces_data = []
+            for trace in traces:
+                trace_dict = trace.to_dict()
+                filtered_trace = filter_json_by_fields(trace_dict, field_list)
+                traces_data.append(filtered_trace)
+            result = {"traces": traces_data, "next_page_token": traces.token}
+        click.echo(json.dumps(result, indent=2))
+    else:
+        # Table output format
+        table = []
+        for trace in traces:
+            trace_dict = trace.to_dict()
+            row = []
+
+            for field in field_list:
+                values = jsonpath_extract_values(trace_dict, field)
+                cell_value = format_table_cell_value(field, None, values)
+                row.append(cell_value)
+
+            table.append(row)
+
+        click.echo(_create_table(table, headers=field_list))
+
+        if traces.token:
+            click.echo(f"\nNext page token: {traces.token}")
+
+
+@commands.command("get")
+@TRACE_ID
+@click.option(
+    "--extract-fields",
+    type=click.STRING,
+    help="Filter and select specific fields using dot notation. "
+    "Examples: 'info.trace_id', 'info.assessments.*', 'data.spans.*.name'. "
+    "Comma-separated for multiple fields. "
+    "If not specified, returns all trace data.",
+)
+@click.option(
+    "--verbose",
+    is_flag=True,
+    help="Show all available fields in error messages when invalid fields are specified.",
+)
+def get_trace(trace_id, extract_fields, verbose):
+    """
+    All trace details will print to stdout as JSON format.
+
+    \b
+    Examples:
+    # Get full trace
+    mlflow traces get --trace-id tr-1234567890abcdef
+
+    \b
+    # Get specific fields only
+    mlflow traces get --trace-id tr-1234567890abcdef \\
+        --extract-fields "info.trace_id,info.assessments.*,data.spans.*.name"
+    """
+    client = TracingClient()
+    trace = client.get_trace(trace_id)
+    trace_dict = trace.to_dict()
+
+    if extract_fields:
+        field_list = [f.strip() for f in extract_fields.split(",")]
+        # Validate fields against trace data
+        try:
+            validate_field_paths(field_list, trace_dict, verbose=verbose)
+        except ValueError as e:
+            raise click.UsageError(str(e))
+        # Filter to selected fields only
+        filtered_trace = filter_json_by_fields(trace_dict, field_list)
+        json_trace = json.dumps(filtered_trace, indent=4)
+    else:
+        # Return full trace
+        json_trace = json.dumps(trace_dict, indent=4)
+
```

Comment:
```suggestion
        json_trace = json.dumps(filtered_trace, indent=2)
    else:
        # Return full trace
        json_trace = json.dumps(trace_dict, indent=2)
```

is there a reason we chose 4?

Reply:
nope, done